### ISSUE TITLE
Insert link accessibility

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#694]
See also #679

All form elements need a properly associated label. As per the [WordPress accessibility standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/#labeling), it must be an explicitly associated label (with for/id attributes, not wrapping the field).

The label can be visually hidden. Also, please consider the placeholder attribute is not a replacement for a label, see https://core.trac.wordpress.org/ticket/40331

Regarding the buttons, at the moment they don't have any text or label that can be announced by assistive technologies, they contain just SVG icons, see #528

These buttons need to be labeled, an `aria-label` attribute or a visually hidden text within the buttons would be appropriate.

when inserting a link:

<img width="935" alt="link insert" src="https://cloud.githubusercontent.com/assets/1682452/25772091/295633ec-3263-11e7-8c79-f9881da85962.png">

when editing a link:

<img width="926" alt="link edit" src="https://cloud.githubusercontent.com/assets/1682452/25772090/2952cc48-3263-11e7-9072-f09c3d1b9875.png">

not sure what to do when editing a link, currently there's no form, just a not-editable link element with an empty `href` attribute. Waiting for development to complete for further review.

**Comment by afercia** [comment]
Looking back at this I see there's now a field to enter the link URL and seems almost ready (no search for now). However, the button to insert a link:

![screen shot 2017-06-07 at 17 53 18](https://user-images.githubusercontent.com/1682452/26894002-685430da-4bbd-11e7-9696-d689a8ad9acf.png)

needs to be labeled (an `aria-label` would work) as well as the edit and unlink buttons:

<img width="577" alt="screen shot 2017-06-07 at 20 32 52" src="https://user-images.githubusercontent.com/1682452/26895148-e77a23e4-4bc0-11e7-87c6-4113ad3db3e7.png">

as there's nothing to announce _what_ these buttons are, they're announced just as "button".

To be 100% compliant, the input field itself needs to be labeled (explicitly associated label, nor wrapping, using for/id attributes) and the label can also be visually hidden (screen-reader-text) if a visible one isn't desirable.

Once a link is inserted in the content, and the cursor on the link makes the toolbar appear, I couldn't find a way to use the Edit and Unlink buttons with a keyboard, unless I'm missing something.

**Comment by mehigh** [comment]
I'm interested in working on this feature and improving the accessibility of the add a link functionality.

**Comment by mehigh** [comment]
Thank you for the suggestions, @afercia 
I've handled the a11y concerns to improve the screen reader experience.

Also, I created a 30s screencast showing how one would arrive to the edit / unlink buttons with the keyboard:
[Screencast showing add / edit link keyboard navigation functionality in Guttenberg](http://files.urldocs.com/share/keyboard-navigation/keyboard-navigation.mp4)

**Comment by afercia** [comment]
@mehigh thanks! I'd just consider to improve a bit the label text and change:

"Submit" to "Apply" or maybe "Insert link"
("Apply" is the text currently used in WordPress)

"Unlink" to "Remove link"
("Unlink" si difficult to translate in some languages, "Remove link" is the current text used in WP)

**Comment by mehigh** [comment]
Thank you. I've updated my PR to include the wording you suggested. 

I picked the common "Apply" to allow existing translations to pass through into this button properly.
